### PR TITLE
Move configuration to php

### PR DESCRIPTION
**Pull request by Zowac**
| Q             | A
| ------------- | ---
| Bug fix?      | no 
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no
| License       | MIT

### CODE REVIEW

**Comment by dunglas** [comment]
This should target master as it's a new feature.

File: src/Bridge/Symfony/Bundle/Resources/config/data_provider.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+use ApiPlatform\Core\DataProvider\CollectionDataProviderInterface;
+use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
+use ApiPlatform\Core\DataProvider\Pagination;
+use ApiPlatform\Core\DataProvider\SubresourceDataProviderInterface;
+use Symfony\Component\DependencyInjection\ServiceLocator;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('api_platform.serializer_locator', ServiceLocator::class)
+            ->args([['serializer' => service('api_platform.serializer')]])
```
Comment (chalasr) [inline src/Bridge/Symfony/Bundle/Resources/config/data_provider.php:26]: You need to import that `service` function (see broken CI).
Symfony core does it by using `namespace Symfony\Component\DependencyInjection\Loader\Configurator` on top of each config file, but I'd rather add a `use function` statement  if it works without namespace.

Comment (Zowac) [reply src/Bridge/Symfony/Bundle/Resources/config/data_provider.php:28]: Change done, thank you :)

File: src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php [hunk 2]
```diff
@@ -89,7 +89,7 @@ public function prepend(ContainerBuilder $container): void
      */
     public function load(array $configs, ContainerBuilder $container): void
     {
-        $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
+        $phpFileLoader = new PhpFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
```
[the remark this answers is not in the document]
Comment (Zowac) [reply src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php]: revert done, thx

File: src/Bridge/Symfony/Bundle/Resources/config/doctrine_orm.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        ->alias(OrderFilter::class, 'api_platform.doctrine.orm.order_filter')
+
+        ->set('api_platform.doctrine.orm.range_filter', RangeFilter::class)
+            ->abstract()
+            ->args([ref('doctrine'), null, ref('logger')->ignoreOnInvalid(), '$nameConverter' => ref('api_platform.name_converter')->ignoreOnInvalid()])
+        ->alias(RangeFilter::class, 'api_platform.doctrine.orm.range_filter')
+
+        ->set('api_platform.doctrine.orm.date_filter', DateFilter::class)
+            ->abstract()
+            ->args([ref('doctrine'), null, ref('logger')->ignoreOnInvalid(), '$nameConverter' => ref('api_platform.name_converter')->ignoreOnInvalid()])
```
Comment (chalasr) [inline src/Bridge/Symfony/Bundle/Resources/config/doctrine_orm.php:85]: The `ref()` function has been renamed to `service()` in 5.1. We need to declare a `service()` proxy function that delegates to the right function internally depending on existence (and remove that proxy when support for symfony/dependency-injection 4.4 gets removed in the future).

File: src/Bridge/Symfony/Bundle/Resources/config/nelmio_api_doc.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<?php
```
Comment (soyuka) [inline src/Bridge/Symfony/Bundle/Resources/config/nelmio_api_doc.php:1]: you can ignore these in phpstan see what we did at https://github.com/api-platform/core/commit/bfca0eb9f064eff70ea570ba18c0326e6dd7291f#diff-6f19df6a6307a48db0940e6897591fb08776d2db8a6134737aa708defdb5c92dR19

Comment (Zowac) [reply src/Bridge/Symfony/Bundle/Resources/config/nelmio_api_doc.php:1]: Done, thx :)

File: src/Bridge/Symfony/Bundle/Resources/function.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+if (!\function_exists(__NAMESPACE__.'\service')) {
```
Comment (chalasr) [inline src/Bridge/Symfony/Bundle/Resources/function.php:16]:
```suggestion
// Forward compatibility layer for the "service()" method.
// To be removed along with support for symfony/dependency-injection <5.1.
if (!\function_exists(__NAMESPACE__.'\service')) {
```

**Review by chalasr — APPROVED** [review]
LGTM 👍

File: src/Bridge/Symfony/Bundle/Resources/function.php (hunk 5 above)
Comment (Zowac) [reply src/Bridge/Symfony/Bundle/Resources/function.php:19]: Comments added, thx :)

File: src/Bridge/Symfony/Bundle/Resources/config/doctrine_orm.php (hunk 3 above)
Comment (Zowac) [reply src/Bridge/Symfony/Bundle/Resources/config/doctrine_orm.php]: Done, thx for you help

File: src/Bridge/Symfony/Bundle/Resources/function.php [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+declare(strict_types=1);
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+// Forward compatibility layer for the "service()" method.
+// To be removed along with support for symfony/dependency-injection <5.1.
+
+if (!\function_exists(__NAMESPACE__.'\service')) {
+    eval('
```
Comment (dunglas) [inline src/Bridge/Symfony/Bundle/Resources/function.php:20]: No way 😁

Either we bump the minimal version of dependency-injection to 5.1, or we keep XML for now and merge this PR in 3.0, but it's too ugly for a very limited benefit (if any) IMHO.

**Review by dunglas — CHANGES_REQUESTED** [review]

File: src/Bridge/Symfony/Bundle/Resources/function.php (hunk 6 above)
Comment (chalasr) [reply src/Bridge/Symfony/Bundle/Resources/function.php:20]: Fair enough :) Let's bump Symfony to 5.1+ and merge this in 3.x then

**Pull request closed without merging**